src/text2.c

```diff
@@ -2463,11 +2467,746 @@ void sub_8007D00(UnkTextStruct1 *a0, u32 a1, s32 x, s32 y, s32 a4, s32 color)
+
+    r2 = strPtr->unk4 * (a3 / 8);
+    r0 = 8 * (r2 + (a2 / 8));
+    r1 = a3 - ((a3 / 8) * 8);
+    r5 = &strPtr->unk18[r0 + r1];
+    while (a5 > 0) {
+        u32 *loopPtr;
+        s32 r4, r6;
+
+        loopPtr = r5;
+        r4 = a4;
+        r6 = a2;
+        if (strPtr->unk3C > r5) {
+            strPtr->unk3C = r5;
+        }
+
+        while (r4 > 0) {
+            s32 r3;
+            if (r4 < 8) {
+                r3 = r4;
+                *loopPtr &= gUnknown_80B8868[(r4 & 7) + 8];
+            }
+            else if ((r6 & 7) != 0) {
+                r3 = 8 - (r6 & 7);
+                *loopPtr &= gUnknown_80B8868[r6 & 7];
+            }
+            else {
+                r3 = 8;
+                *loopPtr = r6 & 7;
+            }
+
+            loopPtr += 8;
+            r4 -= r3;
+            r6 += r3;
+        }
+        a5--;
+        a3++;
+        if (strPtr->unk40 < r5) {
+            strPtr->unk40 = r5;
+        }
+        r5++;
+        if (!(a3 & 7)) {
+            r5 = &r5[strPtr->unk20];
+        }
+    }
+}
+
+void sub_8008818(UnkTextStruct1 *a0, s32 a1, s32 a2, s32 a3, s32 a4, s32 a5);
+
+void sub_80087EC(s32 a0, s32 a1, s32 a2, s32 a3, s32 a4)
+{
+    sub_8008818(gUnknown_2027370, a0, a1, a2, a3, a4);
+}
+
+UNUSED void nullsub_176(void) {}
+
+void sub_8008818(UnkTextStruct1 *a0, s32 a1, s32 a2, s32 a3, s32 a4, s32 a5)
+{
+    s32 i, j;
+    UnkTextStruct1 *strPtr = &a0[a1];
+    s32 a2Div = a2 / 8;
+    s32 a3Div = a3 / 8;
+
+    for (i = 0; i < a5; i += 8) {
+        u32 *ptr = &strPtr->unk18[((strPtr->unk4 * a3Div) + a2Div) * 8];
+        for (j = 0; j < a4; j += 8) {
+            if (strPtr->unk3C > ptr) {
+                strPtr->unk3C = ptr;
+            }
+            *(ptr++) = 0;
+            *(ptr++) = 0;
+            *(ptr++) = 0;
+            *(ptr++) = 0;
+            *(ptr++) = 0;
+            *(ptr++) = 0;
+            *(ptr++) = 0;
+            *ptr = 0;
+            if (strPtr->unk40 < ptr) {
+                strPtr->unk40 = ptr;
+            }
+            ptr++;
+        }
+        a3Div++;
+    }
+}
+
+extern u8 gUnknown_20274A5;
+bool8 xxx_update_bg_vram(UnkTextStruct1 *a0);
+
+bool8 xxx_call_update_bg_vram(void)
+{
+    bool8 ret = FALSE;
+    if (gUnknown_20274A5 != 0) {
+        gUnknown_20274A5 = 0;
+        sub_80099C0();
+    }
+    ret = xxx_update_bg_vram(gUnknown_2027370);
+    return ret;
+}
+
+bool8 xxx_update_bg_vram(UnkTextStruct1 *a0)
+{
+    s32 i, j;
+    u32 r5;
+    bool8 ret = FALSE;
+
+    for (i = 0; i < 4; i++) {
+        UnkTextStruct1 *strPtr = &a0[i];
+        if (strPtr->unk4 == 0)
+            continue;
+        r5 = strPtr->unk38;
+        if (r5 == 0)
+            continue;
+
+        if (strPtr->unk45) {
+            u32 *r2, *r1;
+
+            CpuCopy(strPtr->unk28, strPtr->unk18, 0xD00);
+            // The reason for void casts is because we want to add 0xD00/r5 directly to pointers. Because pointers are u32, without the casts, it would multiply the value by 4.
```

Comment:
This is likely a part of the macro or inline that invokes CpuCopy. It advances the pointers (typed void* or u8*) by the amount of copied bytes.